config.go

```diff
@@ -67,4 +67,6 @@ type State struct {
 	Pid int `json:"pid"`
 	// BundlePath is the path to the container's bundle directory.
 	BundlePath string `json:"bundlePath"`
+	// Labels holds runtime-specified information that is not structured by this specification
+	Labels map[string]json.RawMessage `json:"labels"`
```

Comment:
Labels are typically used to grouping objects. See in docker and kubernetes. 
I'm weary of overloading that concept to hold arbitrary data. 
I'd say lets introduce a separate `Annotation` field that users can use to pass arbitrary data. Is this information expected to be persisted in the `State`?
Also if we switch to protobufs, extensions should let us pass arbitrary data and avoid having to include `Annotations` in the Spec.
Any thoughts @vbatts @philips ?